tensorboard/webapp/hparams/_redux/types.ts

```diff
@@ -0,0 +1,69 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {
+  DiscreteFilter,
+  HparamSpec,
+  IntervalFilter,
+  MetricSpec,
+} from '../_types';
+
+export interface HparamsMetricsAndFilters {
+  hparam: {
+    specs: HparamSpec[];
+    defaultFilters: Map<string, DiscreteFilter | IntervalFilter>;
+  };
+  metric: {
+    specs: MetricSpec[];
+    defaultFilters: Map<string, IntervalFilter>;
+  };
+}
+
+export type ExperimentToHparams = Record<
+  // experiemnt Id.
+  string,
+  HparamsMetricsAndFilters
+>;
+
+/**
+ * Key used to namespace the hparams reducer.
+ */
+export const HPARAMS_FEATURE_KEY = 'hparams';
+
+export interface HparamsState {
+  specs: ExperimentToHparams;
+  /**
+   * RATIONALE: we do not use the routeContextedState because of the following reasons.
```

Comment:
not sure about this but it might be better to be capital R? as other codes show `RouteContextedState`